modules/git/repo_commit_gogit.go

```diff
@@ -50,16 +50,16 @@ func (repo *Repository) ConvertToSHA1(commitID string) (SHA1, error) {
 		}
 	}
 
-	actualCommitID, err := NewCommand(repo.Ctx, "rev-parse", "--verify", commitID).RunInDir(repo.Path)
+	stdout, stderr, err := NewCommand(repo.Ctx, "rev-parse", "--verify", commitID).RunWithContextString(&RunContext{Dir: repo.Path, Timeout: -1})
```

Comment:
`RunWithContextString` is only invoked once here. It's really necessary?

Reply:
See my comment